coreclient/src/contacts/persistence.rs

```diff
@@ -387,16 +384,21 @@ mod tests {
             wai_ear_key: WelcomeAttributionInfoEarKey::random().unwrap(),
             user_profile_base_secret: user_profile_key.base_secret().clone(),
         };
+
+        let mut txn = pool.begin().await?;
+
         let contact = partial
             .mark_as_complete(
-                &pool,
+                &mut txn,
                 &mut store_notifier,
                 friendship_package,
                 AsClientId::new(user_name.clone(), Uuid::new_v4()),
                 user_profile_key.index().clone(),
             )
             .await?;
 
+        txn.commit().await?;
```

Comment:
Should this be `with_transaction`?

Reply:
with_transaction is defined for CoreUser. Can I access it from here?

Reply:
Ah, my bad. Never mind!